version: "3.8"

services:
  contao:
    image: splintnet/dockerized-contao
    volumes:
      - vendor:/application/vendor
      - web:/application/web
      - system:/application/system
      - bundles:/application/bundles
      - assets:/application/assets
      - share:/application/share
      - ./.env:/application/.env
      - ./files:/application/files
      - ./templates:/application/templates
      - ./logs:/application/var/logs
    external_links:
      - db
    networks:
      - db
      - traefik

  nginx:
    image: nginx:alpine
    volumes:
      - ./docker/config/nginx.conf:/etc/nginx/conf.d/default.conf
      - vendor:/application/vendor:ro
      - web:/application/web:ro
      - system:/application/system:ro
      - bundles:/application/bundles:ro
      - assets:/application/assets:ro
      - share:/application/share:ro
      - ./files:/application/files:ro
    depends_on:
      - contao
    networks:
      traefik:
    # labels:
      # - "traefik.enable=true"
      # - "traefik.http.routers.contao.rule=Host(`your-site.com`)"
      # - "traefik.http.routers.contao.tls.certresolver=myhttpchallenge"
      # - "traefik.http.routers.contao.entrypoints=websecure"
      # - "traefik.http.routers.contao.tls=true"

volumes:
  vendor:
  web:
  system:
  bundles:
  assets:
  share:

networks:
  db:
    external: true
  traefik:
    external: true
